geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Stat.scala

```diff
@@ -0,0 +1,197 @@
+/***********************************************************************
+  * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0 which
+  * accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import java.util.Date
+
+import org.joda.time.format.DateTimeFormat
+import org.locationtech.geomesa.utils.stats.MinMaxHelper._
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.util.parsing.combinator.RegexParsers
+
+/**
+ * Stats used by the StatsIterator to compute various statistics server-side for a given query.
+ */
+trait Stat {
+  /**
+   * Compute statistics based upon the given simple feature.
+   * This method will be called for every SimpleFeature a query returns.
+   *
+   * @param sf
+   */
+  def observe(sf: SimpleFeature): Unit
+
+  /**
+   * Meant to be used to combine two Stats of the same subtype.
+   * Used in the "reduce" step client-side.
+   *
+   * @param other the other stat to add (this will always be a stat of the same sub-type unless something horrible happens)
+   * @return the original stat with the other stat's data added
+   */
+  def add(other: Stat): Stat
+
+  /**
+   * Serves as serialization needed for storing the computed statistic in a SimpleFeature.
+   *
+   * @return stat serialized as a json string
+   */
+  def toJson(): String
+
+  /**
+   * Necessary method used by the StatIterator.
+   * Leaving the isEmpty as false ensures that we will always get a stat back from the query
+   * (even if the query doesn't hit any rows).
+   *
+   * @return boolean value
+   */
+  def isEmpty: Boolean = false
+
+  /**
+   * Clears the stat to its original state when first initialized.
+   * Necessary method used by the StatIterator.
+   */
+  def clear(): Unit
+
+  /**
+   * Equals method to simplify testing
+   *
+   * @param obj object to compare
+   * @return true if equal to obj
+   */
+  override def equals(obj: Any): Boolean
```

Comment:
Ah right, misread your comment. Yeah, I put the signature for equals in the trait in an attempt to force stats to have to implement equals, but that didn't work.